src/v/kafka/server/snc_quota_manager.h

```diff
@@ -83,6 +112,12 @@ class snc_quota_manager
     config::binding<double> _kafka_quota_balancer_min_shard_thoughput_ratio;
     config::binding<quota_t> _kafka_quota_balancer_min_shard_thoughput_bps;
 
+    // operational, only used in the balancer shard
+    ss::timer<ss::steady_clock_type> _balancer_timer;
+    ss::steady_clock_type::time_point _balancer_timer_last_ran;
+    ss::gate _balancer_gate;
+    ss::semaphore _balancer_semaphore{1};
```

Comment:
> I don't think the semaphore is necessary because you are controlling the execution of the fiber and you can make assumptions of concurrency. If you only call `arm` once every iteration, then there can't be a case where multiple invocations of the timer cause two fibers to run the algorithm at the same time. So i think this can just be removed.

That's about balancer stability and tolerance to changing load on the reactor: see https://github.com/redpanda-data/redpanda/pull/8335#discussion_r1084358184

Reply:
You're actually right about `quota_balancer_step()` alone: the timer is rearmed after each step is executed, so there should be no implication on concrrency. The semaphore is needed here to prevent concrrency between `quota_balancer_step()` and `quota_balancer_update()`, as well as between potentially concurrent fibers of the latter.

Reply:
> If you did need a semaphore of 1 unit check out `ssx::mutex` instead

I see why I could not find it: it's in `utlis/` not in `ssx/`. However `ssx/` would have been a better fit wouldn't it?